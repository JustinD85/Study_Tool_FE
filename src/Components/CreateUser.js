import React, { Component } from 'react';

class CreateUser extends Component {

  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      email: ' '
    };
  }

  handleSubmit = (event) => {
    const { handleCreateUser, handleChangeView } = this.props;
    event.preventDefault();
    var shouldGoBack = event.target.innerText === "Back";

    if (shouldGoBack) {
      handleChangeView('login')
    } else {
      const { firstName, lastName, email } = this.state;
      handleCreateUser({
        [email]: {
          firstName,
          lastName,
          email,
          score: 0,
          html_elements: { words: [] },
          javascript: { words: [] },
          cascading_style_sheets: { words: [] },
          mod_one_vocab: { words: [] }
        }
      });
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState(
      {
        [name]: value
      }
    );
  }

  render() {
    return (
      <main className="create-user-container">
        <h1>
          Create User Screen
      </h1>
        <form>
          <label>
            First Name
          <input name="firstName" onChange={this.handleChange} />
          </label>
          <label>
            Last Name
          <input name="lastName" onChange={this.handleChange} />
          </label>
          <label>
            Email
          <input name="email" onChange={this.handleChange} />
          </label>
          <button onClick={this.handleSubmit} >Back</button>
          <button type="submit" onClick={this.handleSubmit}>Submit</button>

        </form>
      </main>);
  }
}
export default CreateUser;
