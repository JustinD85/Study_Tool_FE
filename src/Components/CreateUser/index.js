import React, {Component} from 'react';

class CreateUser extends Component{

  constructor({createUser}){
    super();
    this.state = {
      firstName:'',
      lastName:'',
      email:''
    };
    this.createUser = createUser;
  }

  handleSubmit = (event)=>{
    event.preventDefault();
    const {firstName, lastName, email} = this.state;
    this.createUser({
      [email]:{
        firstName,
        lastName,
        email}
    });
  }

  handleChange = (event) =>{
    const {name, value} = event.target;
    this.setState(
      {
        [name]:value
      }
    );
  }

  render(){
  return (
    <main>
      <h1>
        Create User Screen
      </h1>
      <form onSubmit={this.handleSubmit}>
        <label>
          First Name
          <input name="firstName" onChange={this.handleChange}/>
        </label>
        <label>
          Last Name
          <input name="lastName" onChange={this.handleChange} />
        </label>
        <label>
          Email
          <input name="email" onChange={this.handleChange} />
        </label>
        <input type="submit"/>
      </form>
    </main>);
}
}
export default CreateUser;
