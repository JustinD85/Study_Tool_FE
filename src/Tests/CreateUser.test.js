import React from 'react';
import CreateUser from '../Components/CreateUser';
import { shallow } from 'enzyme';

describe('<CreateUser/>', () => {
  let wrapper;
  let handleCreateUserMock = jest.fn();
  let handleChangeViewMock = jest.fn();
  let event = {
    target: {
      innerText: "Back"
    },
    preventDefault: () => { }
  }
  beforeEach(() => {
    wrapper = shallow(
      <CreateUser
        handleChangeView={handleChangeViewMock}
        handleCreateUser={handleCreateUserMock}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to go back to login screen', () => {
    
    wrapper.instance().handleSubmit(event);
    expect(handleChangeViewMock).toHaveBeenCalled();
  });

  it('should be able to create User', () => {
    event.target.innerText = "";
    wrapper.instance().handleSubmit(event);
    expect(handleCreateUserMock).toHaveBeenCalled();
  });

  it('should track user form data entry', () => {
    expect(wrapper.state('firstName')).toEqual('');
    wrapper.instance().handleChange({ target: { name: 'firstName', value: 'Justin' } }  );
    expect(wrapper.state('firstName')).toEqual('Justin');
  })
});