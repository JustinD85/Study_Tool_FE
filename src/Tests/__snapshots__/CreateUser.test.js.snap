// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUser/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser
    handleChangeView={[MockFunction]}
    handleCreateUserMock={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Create User Screen
        </h1>,
        <form>
          <label>
            First Name
            <input
              name="firstName"
              onChange={[Function]}
            />
          </label>
          <label>
            Last Name
            <input
              name="lastName"
              onChange={[Function]}
            />
          </label>
          <label>
            Email
            <input
              name="email"
              onChange={[Function]}
            />
          </label>
          <button
            onClick={[Function]}
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>,
      ],
      "className": "create-user-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create User Screen",
        },
        "ref": null,
        "rendered": "Create User Screen",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              First Name
              <input
                name="firstName"
                onChange={[Function]}
              />
            </label>,
            <label>
              Last Name
              <input
                name="lastName"
                onChange={[Function]}
              />
            </label>,
            <label>
              Email
              <input
                name="email"
                onChange={[Function]}
              />
            </label>,
            <button
              onClick={[Function]}
            >
              Back
            </button>,
            <button
              onClick={[Function]}
              type="submit"
            >
              Submit
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "First Name",
                <input
                  name="firstName"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "First Name",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "firstName",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Last Name",
                <input
                  name="lastName"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Last Name",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "lastName",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email",
                <input
                  name="email"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Back",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Create User Screen
          </h1>,
          <form>
            <label>
              First Name
              <input
                name="firstName"
                onChange={[Function]}
              />
            </label>
            <label>
              Last Name
              <input
                name="lastName"
                onChange={[Function]}
              />
            </label>
            <label>
              Email
              <input
                name="email"
                onChange={[Function]}
              />
            </label>
            <button
              onClick={[Function]}
            >
              Back
            </button>
            <button
              onClick={[Function]}
              type="submit"
            >
              Submit
            </button>
          </form>,
        ],
        "className": "create-user-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create User Screen",
          },
          "ref": null,
          "rendered": "Create User Screen",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                First Name
                <input
                  name="firstName"
                  onChange={[Function]}
                />
              </label>,
              <label>
                Last Name
                <input
                  name="lastName"
                  onChange={[Function]}
                />
              </label>,
              <label>
                Email
                <input
                  name="email"
                  onChange={[Function]}
                />
              </label>,
              <button
                onClick={[Function]}
              >
                Back
              </button>,
              <button
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "First Name",
                  <input
                    name="firstName"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "First Name",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "firstName",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Last Name",
                  <input
                    name="lastName"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Last Name",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "lastName",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email",
                  <input
                    name="email"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Email",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Back",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
